src/javascript/expression.rs

```diff
@@ -147,15 +178,49 @@ impl<'module> Generator<'module> {
     }
 
     fn call<'a>(&mut self, fun: &'a TypedExpr, arguments: &'a [CallArg<TypedExpr>]) -> Output<'a> {
-        let fun = self.not_in_tail_position(|gen| gen.expression(fun))?;
-        let arguments = self.not_in_tail_position(|gen| {
-            call_arguments(
-                arguments
+        match fun {
+            // Short circuit creation of a record so the rendered JS creates the record inline.
+            TypedExpr::Var {
+                constructor:
+                    ValueConstructor {
+                        variant:
+                            ValueConstructorVariant::Record {
+                                name,
+                                arity,
+                                field_map,
+                                ..
+                            },
+                        ..
+                    },
+                ..
+            } => {
+                let tail = self.tail_position;
```

Comment:
I could work out how to use as it is, because I need to return a collection of document pieces to pass to `construct_record` where as `not_in_tail_position` returns a Document.

Reply:
@lpil think all the other issues have been addresses. I guess `not_in_tail_position` could be parameterised to return anything. I think it's simpler this way because it makes the job of `not_in_tail_position` more explicit but I could be convinced either way.

Reply:
Oh yes good point.